pydantic_ai_slim/pydantic_ai/models/fallback.py

```diff
@@ -62,7 +66,11 @@ async def request(
         for model in self.models:
             try:
                 response, usage = await model.request(messages, model_settings, model_request_parameters)
-                response.model_used = model  # type: ignore
+                span = get_current_span()
+                if span.is_recording():
+                    attributes = getattr(span, 'attributes', {})
+                    if attributes.get('gen_ai.request.model') == self.model_name:
+                        span.set_attributes(InstrumentedModel.model_attributes(model))
```

Comment:
I feel like this should be outside this try-except. Like, errors with otel shouldn't trigger fallback, it means something else is wrong